src/java.base/share/classes/java/security/doc-files/security-related-system-properties.html

```diff
@@ -0,0 +1,199 @@
+<!doctype html>
+<!--
+ Copyright (c) 2025, Oracle and/or its affiliates. All rights reserved.
+ DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+
+ This code is free software; you can redistribute it and/or modify it
+ under the terms of the GNU General Public License version 2 only, as
+ published by the Free Software Foundation.  Oracle designates this
+ particular file as subject to the "Classpath" exception as provided
+ by Oracle in the LICENSE file that accompanied this code.
+
+ This code is distributed in the hope that it will be useful, but WITHOUT
+ ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ version 2 for more details (a copy is included in the LICENSE file that
+ accompanied this code).
+
+ You should have received a copy of the GNU General Public License version
+ 2 along with this work; if not, write to the Free Software Foundation,
+ Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+
+ Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ or visit www.oracle.com if you need additional information or have any
+ questions.
+-->
+<html lang="en">
+<head>
+    <meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
+    <title>Security-Related System Property</title>
+    <style>
+        #debug {
+         border: 1px solid black;
+         border-collapse: collapse;
+         margin: 0 auto;
+        }
+        #debug caption {
+          font-weight: bold;
+          font-size: smaller;
+        }
+        #debug, #debug th, #debug td {
+         border: 1px solid black;
+         padding: 2px .5em;
+        }
+        #debug tbody th {
+         font-weight: normal;
+         text-align:left;
+        }
+    </style>
+</head>
+<body LANG="en-US" DIR="LTR">
+<H1 style="text-align:center">Security-Related System Property</H1>
+<p>
+    There are several system properties used to
+    alter the mechanisms and behavior of the various classes of the
+    <code>java.security</code> package. Some are checked only once at startup of the VM,
+    and therefore are best set using the <code>-D</code> option of the java command,
+    while others have a more dynamic nature and can also be changed using
+    the <a href="../../lang/System.html#setProperty(java.lang.String,java.lang.String)">System.setProperty()</a> API.
+    </P>
+<a id="security.debug"></a>
+<H2>Debug</H2>
+<p><B>{@systemProperty java.security.debug}</B><BR></p>
+<H2>Printing Thread and Timestamp Information</H2>
+<p>
+    You can append the following strings to the value specified in the <code>java.security.debug</code>
+    system property to print additional information:
+<ul>
+    <li><code>+thread:</code>Print thread and caller information</li>
```

Comment:
Put `:` outside `<code>`.

Hi @coffeys and @seanjmullan, now that we have a formal format for options and sub-options, shall we have one for these modifiers? Should it always be appended after the sub-option(s)? Like
```
x509:ava+thread,certpath:ocsp,verbose+timestamp+thread
```
Also, how do we specify multiple engines in `provider`? Should it look like this?
```
provider:engine=Cipher,Mac
```

Reply:
I'd probably have to point to the CSR that introduced the +thread, +timestamp options :
https://bugs.openjdk.org/browse/JDK-8327569

The "+" symbol should directly follow a valid debug option to be legitimate IMO. It's probably the most intuitive assumption also. 

note that I'd like to switch on `timestamp `and `thread `options by default in a JDK feature release (maybe JDK 25)
They made more sense for the backport (JDK Update) release lines where addition of such meta data might have caused issue for some frameworks parsing this data.

I just logged https://bugs.openjdk.org/browse/JDK-8350689 to track that effort.

Reply:
Anyone still cares about this comment?

I've tried `-Djava.security.debug=certpath:ocsp,verbose+timestamp` (which follows the grammar in this PR) and there is no timestamp shown. Obviously, it's broken into `certpath:ocsp` and `verbose+timestamp` at https://github.com/openjdk/jdk/blob/940aa7c4cf1bf770690660c8bb21fb3ddc5186e4/src/java.base/share/classes/sun/security/util/Debug.java#L192 and the `+timestamp` modifier is not applied to the `certpath` option.

Reply:
I think we should look into it. I think we can handle this as a follow-on bug so that we keep this issue just about the docs.

Reply:
I’m not suggesting any source changes, but this documentation update explicitly states:
> Sub-options are specified by appending a ":" to the option, followed by a list of one more sub-options separated by a comma.
This breaks existing usages. In fact, unless (as Seán C suggested) we drop the 2 modifiers entirely and make them always on, I don’t see a good way to fix it later if we keep the current grammar.

Reply:
Well we are dealing with a syntax that was never specified and is very loosely defined. Either we remove some of the syntax and leave it undefined, or we make sure the syntax works and add more constraints or delimiters, which could break existing usages. But I'm also not sure how much we need to really accommodate corner cases.